### PR TITLE
Chore segment tracking service

### LINKED ISSUE #206: Add Segment Rasa Event Broker
To integrate segment (tracking service):

> We can implement the solution we did in Beautiful-Mind. It is a Rasa Event Broker.
> 1. You can just copy-paste what is in segment.py,
> 2. and put this in the Rasa config:
> event_broker:
>   type: rasa_stack.segment.SegmentSource
> With the correct path.
> 3. Then you just need to add SEGMENT_WRITE_KEY in the helm secrets.

@lp-noel will create a new segment source
Poke @mbelang to create the helm secret

### PR DESCRIPTION
**Pull request by mathieubergeron**
## Description
Added segment source event broker, inspired by https://github.com/dialoguemd/beautiful-mind/blob/master/rasa-stack/rasa_stack/segment.py

To simplify development in local, and contrary to beautiful-mind, core startup will not fail if `SEGMENT_WRITE_KEY` environement variable is not set, but segment source event broker will be disabled.

## Related issues
closes #206 

## Checklist

- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [x] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: core/Dockerfile [hunk 1]
```diff
@@ -22,5 +22,6 @@ USER 1001
 COPY credentials.yml endpoints.yml ./
 COPY models models
 COPY channels channels
```
Comment (mathieubergeron) [inline core/Dockerfile:24]: `channels` should be moved within `core`.

File: core/core/segment.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# Taken (and adapted) from: https://github.com/dialoguemd/beautiful-mind/blob/master/rasa-stack/rasa_stack/segment.py
+
+
+class SegmentSource(EventBroker):
+    """ Track events with Segment"""
+
+    def __init__(self, write_key: Optional[str] = None):
+
+        if write_key is None:
+            write_key = os.environ.get("SEGMENT_WRITE_KEY")
```
Comment (mathieubergeron) [inline core/core/segment.py:29]: I've notified Maxime about that secret/environment variable - it will be added soon.

Comment (mathieubergeron) [reply core/core/segment.py:29]: (I will rebase by branch once he has added the secret, and check with him that he sees the events in segment.io)

Comment (mathieubergeron) [reply core/core/segment.py:29]: Done

File: core/core/segment.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                # removing the metadata because it sends all of the slots to Segment
+                # event sending only non-NULL slots would eventually create 1000s of columns
+                event.pop("metadata", None)
+            analytics.track(
+                user_id=sender_id,
+                event=event_name,
+                properties=event,
+                context={
+                    # TODO: update this once a versioning strategy is established
+                    "app": {"name": "covidflow", "version": "0.0.0", "build": "0"}
```
Comment (mathieubergeron) [inline core/core/segment.py:65]: @lp-noel is "covidflow" ok ?

Comment (fparga) [inline core/core/segment.py:65]: Is Dialogue OK with this?

Comment (lp-noel) [reply core/core/segment.py:65]: yes

**Comment by lp-noel** [comment]
I was able to setup a Dev source for this, @mathieubergeron can you run through a couple a patients on dev (with this PR deployed to canonical) so we can verify if the data flows correctly?

**Pull request merged**